src/file.rs

```diff
@@ -170,3 +201,29 @@ impl File {
         self.partition_uuid.non_zero()
     }
 }
+impl Debug for File {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        let mut x = f.debug_struct("File");
+        x.field("revision", &self.revision());
+        x.field("addr", &self.addr());
+        x.field("size", &self.size());
+
+        match core::str::from_utf8(self.path()) {
+            Ok(path) => x.field("path", &path),
+            Err(err) => x.field("path", &err),
+        };
+
+        x.field("cmdline", &self.cmdline());
+        x.field("media_type", &self.media_type());
+        x.field("tftp_ip", &self.tftp_ip());
+        x.field("tftp_port", &self.tftp_port());
+        #[cfg(feature = "ipaddr")]
+        x.field("tftp_addr", &self.tftp_addr());
```

Comment:
Instead of having a separate `tftp_addr` field, you could just print `tftp_addr()` if the feature is enabled.